### PR TITLE
Add support for ea/ga access

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
In scope of this pull request we add logic to install ga and ea java versions
### What is not yet completed
-  [x] Setup java from jdkFile.
-  [x] Install ga/ga java versions
-  [ ] Cover new logic with tests
-  [ ] Move preinstalled java versions to the toolcache directory

### CODE REVIEW

File: src/distributors/base-installer.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export abstract class JavaBase {
   protected version: semver.Range;
   protected arch: string;
   protected javaPackage: string;
+  protected access: string;
```
Comment (maxim-lobanov) [inline src/distributors/base-installer.ts:18]: Let's rename `access` to `stable` (true, false)

File: src/distributors/base-installer.ts [hunk 2]
```diff
@@ -85,7 +88,8 @@ export abstract class JavaBase {
   }
 
   // this function validates and parse java version to its normal semver notation
-  protected normalizeVersion(version: string): semver.Range {
+  protected normalizeVersion(version: string) {
+    let access = 'ga';
```
Comment (maxim-lobanov) [inline src/distributors/base-installer.ts:92]: Let's rename access to stable (true, false)
And i think we should resolve boolean stable to `ea` or `ga` later in getAvailableVersions method.

File: src/distributors/base-installer.ts [hunk 3]
```diff
@@ -104,6 +108,7 @@ export abstract class JavaBase {
       if (version[0] >= '0' && version[0] <= '9') {
         version = '>=' + version;
       }
+      access = 'ea';
```
Comment (maxim-lobanov) [inline src/distributors/base-installer.ts:111]: I think we should get rid of `else if (version.split('.').length < 3) {`
Also condition `if (version.endsWith('-ea')) {` can be simplified

File: src/distributors/base-models.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import * as semver from 'semver';
```
Comment (maxim-lobanov) [inline src/distributors/base-models.ts:1]: remove it

Comment (dmitry-shibanov) [reply src/distributors/base-models.ts]: Done. Thank you.

File: src/distributors/base-installer.ts (hunk 3 above)
Comment (dmitry-shibanov) [reply src/distributors/base-installer.ts]: Done. Thank you.

File: src/distributors/base-installer.ts (hunk 2 above)
Comment (dmitry-shibanov) [reply src/distributors/base-installer.ts]: Done. Thank you.

File: src/distributors/base-installer.ts (hunk 1 above)
Comment (dmitry-shibanov) [reply src/distributors/base-installer.ts]: Done. Thank you.

File: dist/setup/index.js [hunk 5]
```diff
@@ -13359,7 +13359,11 @@ class AdoptOpenJDKDistributor extends base_installer_1.JavaBase {
             const jvmImpl = 'hotspot';
             const versionRange = '[1.0,100.0]';
             const encodedVersionRange = encodeURI(versionRange);
+            let release_type = 'ga';
```
Comment (maxim-lobanov) [inline dist/setup/index.js:13362]: nitpick: `release_type` -> `releaseType`
nitpick2: `const releaseType = this.stable ? 'ga' : 'ea'`

File: src/distributors/adoptopenjdk-installer.ts [hunk 6]
```diff
@@ -84,8 +84,14 @@ export class AdoptOpenJDKDistributor extends JavaBase {
     const jvmImpl = 'hotspot';
     const versionRange = '[1.0,100.0]';
     const encodedVersionRange = encodeURI(versionRange);
+    let release_type = 'ga';
```
Comment (maxim-lobanov) [inline src/distributors/adoptopenjdk-installer.ts:87]: nitpick: release_type -> releaseType
nitpick2: const releaseType = this.stable ? 'ga' : 'ea'

File: src/distributors/zulu-installer.ts [hunk 7]
```diff
@@ -77,6 +77,11 @@ export class ZuluDistributor extends JavaBase {
     const platform = this.getPlatformOption();
     const extension = getDownloadArchiveExtension();
     const javafx = features?.includes('fx') ?? false;
+    let release_status = 'ga';
```
Comment (maxim-lobanov) [inline src/distributors/zulu-installer.ts:80]: nitpick: release_type -> releaseType
nitpick2: const releaseType = this.stable ? 'ga' : 'ea'

File: src/distributors/base-installer.ts [hunk 8]
```diff
@@ -11,14 +11,17 @@ export abstract class JavaBase {
...
 
   constructor(protected distributor: string, installerOptions: JavaInstallerOptions) {
     this.http = new httpm.HttpClient('setup-java', undefined, {
       allowRetries: true,
       maxRetries: 3
     });
 
-    this.version = this.normalizeVersion(installerOptions.version);
+    const javaVersionOptions = this.normalizeVersion(installerOptions.version);
+    this.version = javaVersionOptions.version;
```
Comment (maxim-lobanov) [inline src/distributors/base-installer.ts:23]: May be 
```
({version: this.version stable: this.stable} = this.normalizeVersion(installerOptions.version));
```
Or do you think that your approach is more clear?

File: dist/setup/index.js (hunk 5 above)
Comment (dmitry-shibanov) [reply dist/setup/index.js:13362]: Done. Thank you.

File: src/distributors/adoptopenjdk-installer.ts (hunk 6 above)
Comment (dmitry-shibanov) [reply src/distributors/adoptopenjdk-installer.ts]: Done. Thank you.

File: src/distributors/zulu-installer.ts (hunk 7 above)
Comment (dmitry-shibanov) [reply src/distributors/zulu-installer.ts]: Done. Thank you.

File: src/distributors/base-installer.ts (hunk 8 above)
Comment (dmitry-shibanov) [reply src/distributors/base-installer.ts]: Done. Thank you.

**Review by maxim-lobanov — APPROVED** [review]

File: src/distributors/adoptopenjdk-installer.ts [hunk 9]
```diff
@@ -84,8 +84,10 @@ export class AdoptOpenJDKDistributor extends JavaBase {
     const jvmImpl = 'hotspot';
     const versionRange = '[1.0,100.0]';
     const encodedVersionRange = encodeURI(versionRange);
+    let releaseType = this.stable ? 'ga' : 'ea';
```
Comment (vsafonkin) [inline src/distributors/adoptopenjdk-installer.ts:87]: I believe, you can use `const` variable

File: src/distributors/base-installer.ts [hunk 10]
```diff
@@ -11,14 +11,17 @@ export abstract class JavaBase {
...
+  protected stable: boolean;
 
   constructor(protected distributor: string, installerOptions: JavaInstallerOptions) {
     this.http = new httpm.HttpClient('setup-java', undefined, {
       allowRetries: true,
       maxRetries: 3
     });
 
-    this.version = this.normalizeVersion(installerOptions.version);
+    ({ version: this.version, stable: this.stable } = this.normalizeVersion(
```
Comment (vsafonkin) [inline src/distributors/base-installer.ts:22]: Why it needs extra round brackets?

**Review by vsafonkin — APPROVED** [review]

File: src/distributors/base-installer.ts (hunk 10 above)
Comment (maxim-lobanov) [reply src/distributors/base-installer.ts:22]: It is required by JS if you use object {} on left side of =

**Review by AlenaSviridenko — APPROVED** [review]

File: src/distributors/adoptopenjdk-installer.ts (hunk 9 above)
Comment (dmitry-shibanov) [reply src/distributors/adoptopenjdk-installer.ts]: Done. Thank you.

**Pull request merged**